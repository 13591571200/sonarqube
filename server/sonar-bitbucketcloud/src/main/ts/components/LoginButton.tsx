/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import Button, { ButtonAppearances } from '@atlaskit/button';
import { getBaseUrl } from '@sqcore/helpers/urls';

interface Props {
  appearance: ButtonAppearances;
  children: React.ReactNode;
  icon?: JSX.Element;
  onReload: () => void;
  style?: React.CSSProperties;
  sessionUrl: string;
}

export default class LoginButton extends React.PureComponent<Props> {
  handleLoginClick = (event: React.SyntheticEvent<MouseEvent>) => {
    event.preventDefault();
    event.stopPropagation();

    (window as any).authenticationDone = () => {
      this.props.onReload();
    };

    const returnTo = encodeURIComponent('/integration/bitbucketcloud/after_login');
    window.open(
      `${getBaseUrl()}/${this.props.sessionUrl}?return_to=${returnTo}`,
      'Login on SonarCloud',
      'toolbar=0,status=0,width=1100,height=640'
    );
  };

  render() {
    return (
      <Button
        appearance={this.props.appearance}
        iconBefore={this.props.icon}
        onClick={this.handleLoginClick}>
        {this.props.children}
      </Button>
    );
  }
}
