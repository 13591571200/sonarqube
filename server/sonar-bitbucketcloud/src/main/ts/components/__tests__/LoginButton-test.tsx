/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import LoginButton from '../LoginButton';

it('should display correctly as a button with icon', () => {
  const wrapper = shallow(
    <LoginButton
      appearance="primary"
      icon={<i className="my-icon" />}
      onReload={jest.fn()}
      sessionUrl={`sessions/url`}>
      Connect
    </LoginButton>
  );
  expect(wrapper).toMatchSnapshot();
});

it('should correctly open a popup', () => {
  const onReload = jest.fn();
  const open = jest.fn();
  (global as any).open = open;

  const wrapper = shallow(
    <LoginButton appearance="link" onReload={onReload} sessionUrl={`sessions/url`}>
      Connect
    </LoginButton>
  );
  expect(wrapper).toMatchSnapshot();
  (wrapper.find('WithAnalyticsContext').prop('onClick') as Function)({
    preventDefault: () => {},
    stopPropagation: () => {}
  });
  expect(open).toHaveBeenCalledWith(
    '/sessions/url?return_to=%2Fintegration%2Fbitbucketcloud%2Fafter_login',
    'Login on SonarCloud',
    'toolbar=0,status=0,width=1100,height=640'
  );
});
