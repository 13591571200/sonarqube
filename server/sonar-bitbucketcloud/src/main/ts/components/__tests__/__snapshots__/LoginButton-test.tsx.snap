// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly open a popup 1`] = `
<WithAnalyticsContext(WithAnalyticsEvents(WithDeprecationWarnings(Button)))
  analyticsContext={Object {}}
  appearance="link"
  onClick={[Function]}
>
  Connect
</WithAnalyticsContext(WithAnalyticsEvents(WithDeprecationWarnings(Button)))>
`;

exports[`should display correctly as a button with icon 1`] = `
<WithAnalyticsContext(WithAnalyticsEvents(WithDeprecationWarnings(Button)))
  analyticsContext={Object {}}
  appearance="primary"
  iconBefore={
    <i
      className="my-icon"
    />
  }
  onClick={[Function]}
>
  Connect
</WithAnalyticsContext(WithAnalyticsEvents(WithDeprecationWarnings(Button)))>
`;
